crates/sui-rpc-loadgen/src/payload/rpc_command_processor.rs

```diff
@@ -210,66 +211,176 @@ pub async fn check_transactions(
     digests: &[TransactionDigest],
     cross_validate: bool,
+    verify_objects: bool,
+) {
+    let read = clients.read().await;
+    let cloned_clients = read.clone();
+
+    let transactions = join_all(
+        cloned_clients
+            .iter()
+            .enumerate()
+            .map(|(i, client)| async move {
+                let start_time = Instant::now();
+                let transactions = client
+                    .read_api()
+                    .multi_get_transactions_with_options(
+                        digests.to_vec(),
+                        SuiTransactionResponseOptions::full_content(), // todo(Will) support options for this
+                    )
+                    .await;
+                let elapsed_time = start_time.elapsed();
+                debug!(
+                    "MultiGetTransactions Request latency {:.4} for rpc at url {i}",
+                    elapsed_time.as_secs_f64()
+                );
+                transactions
+            }),
+    )
+    .await;
+
+    // TODO: support more than 2 transactions
+    if cross_validate && transactions.len() == 2 {
+        if let (Some(t1), Some(t2)) = (transactions.get(0), transactions.get(1)) {
+            let first = match t1 {
+                Ok(vec) => vec.as_slice(),
+                Err(err) => {
+                    error!("Error unwrapping first vec of transactions: {:?}", err);
+                    error!("Logging digests, {:?}", digests);
+                    return;
+                }
+            };
+            let second = match t2 {
+                Ok(vec) => vec.as_slice(),
+                Err(err) => {
+                    error!("Error unwrapping second vec of transactions: {:?}", err);
+                    error!("Logging digests, {:?}", digests);
+                    return;
+                }
+            };
+
+            cross_validate_entities(digests, first, second, "TransactionDigest", "Transaction");
+
+            if verify_objects {
+                // todo: this can be written better
+                for response in first {
+                    let object_changes: Vec<ObjectID> = response
+                        .object_changes
+                        .clone()
+                        .unwrap()
+                        .iter()
+                        .filter_map(get_object_id)
+                        .collect();
+                    check_objects(clients, object_changes.as_slice(), cross_validate).await;
+                }
+                for response in second {
+                    let object_changes: Vec<ObjectID> = response
+                        .object_changes
+                        .clone()
+                        .unwrap()
+                        .iter()
+                        .filter_map(get_object_id)
+                        .collect();
+                    check_objects(clients, object_changes.as_slice(), cross_validate).await;
+                }
+            }
+        }
```

Comment:
In most cases, these two are the same, can we just fetch the union? Otherwise this will make the script a lot slower to run

Reply:
Thanks that's a good call